Example/PaymentSheet Example/PaymentSheet Example/ExampleEmbeddedElementCheckoutViewController.swift

```diff
@@ -0,0 +1,391 @@
+    @IBOutlet weak var totalLabel: UILabel!
+    @IBOutlet weak var subscribeSwitch: UISwitch!
+
+    var embeddedPaymentElement: EmbeddedPaymentElement!
+
+    private let backendCheckoutUrl = URL(string: baseUrl + "/checkout")!
+    private let confirmIntentUrl = URL(string: baseUrl + "/confirm_intent")!
+    private let computeTotalsUrl = URL(string: baseUrl + "/compute_totals")!
+
+    private struct ComputedTotals: Decodable {
+        let subtotal: Double
+        let tax: Double
+        let total: Double
+    }
+
+    private var computedTotals: ComputedTotals!
+
+    // MARK: - Create an IntentConfiguration
+    private var intentConfig: PaymentSheet.IntentConfiguration {
+        return .init(mode: .payment(amount: Int(computedTotals.total),
+                                    currency: "USD",
+                                    setupFutureUsage: subscribeSwitch.isOn ? .offSession : nil)
+        ) { [weak self] paymentMethod, shouldSavePaymentMethod, intentCreationCallback in
+            Task {
+                do {
+                    // Create and confirm an intent on your server and invoke `intentCreationCallback` with the client secret or an error.
+                    // TODO(yuki): Show client-side confirm, not server-side confirm.
+                    guard let self else {
+                        intentCreationCallback(.failure(ExampleError()))
+                        return
+                    }
+                    let clientSecret = try await self.confirmIntent(paymentMethodID: paymentMethod.stripeId, shouldSavePaymentMethod: shouldSavePaymentMethod)
+                    intentCreationCallback(.success(clientSecret))
+                } catch {
+                    intentCreationCallback(.failure(error))
+                }
+            }
+        }
+    }
+
+    private var currencyFormatter: NumberFormatter {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        formatter.maximumFractionDigits = 2
+        return formatter
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        buyButton.addTarget(self, action: #selector(didTapCheckoutButton), for: .touchUpInside)
+        buyButton.isEnabled = false
+
+        paymentMethodButton.addTarget(self, action: #selector(didTapPaymentMethodButton), for: .touchUpInside)
+        paymentMethodButton.isEnabled = false
+
+        hotDogStepper.isEnabled = false
+        saladStepper.isEnabled = false
+        subscribeSwitch.isEnabled = false
+
+        Task {
+            await self.loadCheckout()
+        }
+    }
+
+    @objc
+    func didTapPaymentMethodButton() {
+        let paymentMethodsViewController = PaymentMethodsViewController(embeddedPaymentElement: embeddedPaymentElement)
+        let navController = UINavigationController(rootViewController: paymentMethodsViewController)
+        present(navController, animated: true)
+    }
+
+    @objc
+    func didTapCheckoutButton() async {
```

Comment:
Surprised this works, didn't think you could use @objc + async on a function definition.

Reply:
Me too!  This is actually untested, buy button won't work until we implement `paymentOption`.  So it might crash at runtime :D